src/ansys/motorcad/core/geometry.py

```diff
@@ -509,7 +518,47 @@ def mirror(self, mirror_line):
                     2 * d * mirror_line.gradient - self.y + 2 * mirror_line.y_intercept,
                 )
         else:
-            raise Exception("Coordinate can only be mirrored about Line()")
+            raise Exception("Coordinate can only be mirrored about Line")
+
+    def rotate(self, centre_point, angle):
+        """Rotate Coordinate around a point for a given angle.
+
+        Parameters
+        ----------
+        centre_point : Coordinate
+            point to rotate Coordinate around.
+        angle : float
+            Angle of rotation in degrees.
+
+        Returns
+        -------
+        Coordinate
+        """
+        angle_r = radians(angle)
+        x_new = centre_point.x + (
+            (cos(angle_r) * (self.x - centre_point.x)) + (-sin(angle_r) * (self.y - centre_point.y))
+        )
+        y_new = centre_point.y + (
+            (sin(angle_r) * (self.x - centre_point.x)) + (cos(angle_r) * (self.y - centre_point.y))
+        )
+
+        return Coordinate(x_new, y_new)
```

Comment:
@matthewAnsys 

Decided to return new coordinate instead of mutating existing.

Not sure I like this but I think mutating coordinate is likely to confuse beginner Python users

Reply:
I personally agree but trying to avoid confusion with beginners.

```python

# mutating correct way
p1 = Coordinate()
p2 = copy(p1)
p2.rotate(90)

# suspect most people will try
p1 = Coordinate()
p2 = p1.rotate(90)
# p1 is p2 (confusing?)

# vs

p1 = Coordinate()
p2 = p1.rotate(90)

# p1 and p2 are different objects - works as expected
```